source/Irrlicht/COpenGLDriver.cpp

```diff
@@ -1909,10 +1909,26 @@ void COpenGLDriver::SAuxContext::flushState(GL_STATE_BITS stateBits)
     if (stateBits & GSB_PIPELINE_AND_RASTER_PARAMETERS)
     {
         if (nextState.pipeline != currentState.pipeline) {
-            //TODO find pipeline GL name in cache and bind
-            //COpenGLExtensionHandler::extGlBindProgramPipeline(nextState.pipeline);
+            SGraphicsPipelineHash hash;
+            for (uint32_t i = 0u; i < COpenGLRenderpassIndependentPipeline::SHADER_STAGE_COUNT; ++i)
+            {
+                hash[i] = currentState.pipeline->getShaderAtIndex(i) ?
+                    static_cast<const COpenGLSpecializedShader*>(currentState.pipeline->getShaderAtIndex(i))->getGLnameForCtx(this->ID) :
+                    0u;
+            }
+
+            GLuint GLname = 0u;
+            auto found = GraphicsPipelineMap.find(hash);
+            if (found != GraphicsPipelineMap.end())
+                GLname = found->second;
+            else
+                GLname = createGraphicsPipelineInCache(hash);
+
+            extGlUseProgram(0);//TODO hm??
```

Comment:
just make sure we never glUseProgram and we'll be ok.

Reply:
glUseProgram we'll be used for compute pipelines

Reply:
then only set to 0 if compute pipeline was used before after last flush.

Reply:
and?

Reply:
status?

Reply:
when COpenGL compute pipelines come

Reply:
@Crisspl want compute pipelines in this PR